cmd/plakar/subcommands/sync/sync.go

```diff
@@ -214,92 +219,262 @@ func (cmd *Sync) Execute(ctx *appcontext.AppContext, repo *repository.Repository
 	return 0, nil
 }
 
-func synchronize(srcRepository *repository.Repository, dstRepository *repository.Repository, snapshotID objects.MAC) error {
-	srcSnapshot, err := snapshot.Load(srcRepository, snapshotID)
-	if err != nil {
+func push(src *snapshot.Snapshot, dst *snapshot.Snapshot, mac objects.MAC, rtype resources.Type, data []byte) (bool, []byte, error) {
+	var err error
+
+	if dst.BlobExists(rtype, mac) {
+		return true, nil, nil
+	}
+
+	if data == nil {
+		data, err = src.GetBlob(rtype, mac)
+		if err != nil {
+			return false, nil, err
+		}
+	}
+	return false, data, dst.PutBlob(rtype, mac, data)
+}
+
+func syncObject(src *snapshot.Snapshot, dst *snapshot.Snapshot, mac objects.MAC) error {
+	found, objbytes, err := push(src, dst, mac, resources.RT_OBJECT, nil)
```

Comment:
I think resources.RT_OBJECT should be pushed after the chunks that constitute it so that an object can't exist on the other end with an incomplete chunks set, no ?

Reply:
fixed